pkg/operator/operator.go

```diff
@@ -743,3 +759,108 @@ func (o *Operator) Config(ctx context.Context, key string) (*manifests.Config, e
 
 	return c, nil
 }
+
+// Upgradeable verifies whether the operator can be upgraded or not. It returns
+// the ConditionStatus with optional reason and message.
+func (o *Operator) Upgradeable(ctx context.Context) (configv1.ConditionStatus, string, string, error) {
+	if !o.lastKnowInfrastructureConfig.HighlyAvailableInfrastructure() {
+		return configv1.ConditionTrue, "", "", nil
+	}
+
+	workloadsCorrectlySpread, reason, message, err := o.WorkloadsCorrectlySpread(ctx)
+	if err != nil {
+		return configv1.ConditionUnknown, "", "", err
+	}
+
+	if !workloadsCorrectlySpread {
+		return configv1.ConditionFalse, reason, message, nil
+	}
+
+	return configv1.ConditionTrue, reason, message, nil
+}
+
+// workloadCorrectlySpread returns whether the selected pods are spread across
+// different nodes ensuring proper high-availability.
+func (o *Operator) workloadCorrectlySpread(ctx context.Context, namespace string, sel map[string]string) (bool, error) {
+	podList, err := o.client.ListPods(ctx, namespace, metav1.ListOptions{LabelSelector: labels.FormatLabels(sel)})
+	if err != nil {
+		return false, err
+	}
+
+	// Skip the check if we can't get enough pods. This prevents setting the status when the cluster is degraded.
+	if len(podList.Items) <= 1 {
+		return true, nil
+	}
+
+	nodes := make(map[string]struct{}, len(podList.Items))
+	for _, pod := range podList.Items {
+		nodes[pod.Spec.NodeName] = struct{}{}
+	}
+
+	return len(nodes) > 1, nil
+}
+
+func (o *Operator) WorkloadsCorrectlySpread(ctx context.Context) (bool, string, string, error) {
+	type workload struct {
+		namespace     string
+		name          string
+		labelSelector map[string]string
+	}
+
+	workloads := []workload{
+		{
+			namespace:     o.namespace,
+			name:          "prometheus-k8s",
+			labelSelector: map[string]string{"app.kubernetes.io/name": "prometheus"},
+		},
+		// TODO: verify correct spreading of Alertmanager pods once we deploy
```

Comment:
This case is handled by the current implementation since we are checking on how many nodes the pods are. Even if 2 replicas of Alertmanager were on the same node, the total number of nodes on which Alertmanager pods are would be 2, so the [check](https://github.com/openshift/cluster-monitoring-operator/pull/1330/files#diff-99c0290799daf9abc6240df64063e20bfaf67b371577b67ac7eec6f4725622ffR800) would return that the workload is correctly spread.

Reply:
Updated in 33a4e83f88a78d2c8e1cd6506495159553773789

Reply:
:+1: